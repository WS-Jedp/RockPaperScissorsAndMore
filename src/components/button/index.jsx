import React from 'react'
import { ButtonSecondaryWrapper, ButtonPrimaryWrapper } from './styles'

export const ButtonSecondary = ({title = 'Rules', action = () => {}}) => (
  <ButtonSecondaryWrapper onClick={action}>
    {title}
  </ButtonSecondaryWrapper>
)

export const ButtonPrimary = ({title = 'Play Again', action = () => {}}) => (
  <ButtonPrimaryWrapper onClick={action}>
    {title}
  </ButtonPrimaryWrapper>
)
