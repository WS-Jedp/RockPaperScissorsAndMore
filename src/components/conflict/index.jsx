import React from 'react'
import { ConflictWrapper } from './styles'

import { PickedOption } from '../pickedOption'
import { PlayAgain } from '../playAgain'

export const Conflict = ({userOption = {}, homeOption = {}}) => {

  const isMobile = false
  
  return (
    <>
      <ConflictWrapper>
        <PickedOption title='You Picked' option={userOption.name} color={userOption.color} picked={true} />
        {
        isMobile && <PlayAgain title='You Win'action={() => {}} />
        }
        <PickedOption title='The House Picked' option={homeOption.name} color={homeOption.color} />
      </ConflictWrapper>
      {
        !isMobile && <PlayAgain title='You Win'action={() => {}} />
      }
    </>
  )
}
