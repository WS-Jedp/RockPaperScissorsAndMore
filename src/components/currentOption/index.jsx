import React from 'react'
import { IconLizard, IconPaper, IconScissors, IconRock, IconSpock } from '../icons'

export const CurrentOption = ({option = 'rock'}) => {

  const optionName = option.toLowerCase(); 

    switch (optionName) {
      case "rock":
        return <IconRock />
        break;
      case "paper":
        return <IconPaper />
        break;
      case "scissors":
    return <IconScissors />
        break;
      case "lizard":
        return <IconLizard />
        break;
      case "spock":
        return <IconSpock />
        break;
    
      default:
        return <IconRock />
        break;
    }
  }