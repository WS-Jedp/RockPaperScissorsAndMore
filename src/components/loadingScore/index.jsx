import React from 'react'
import { LoadingWrapper } from './styles'

export const LoadingScore = () => (
  <LoadingWrapper />
)
