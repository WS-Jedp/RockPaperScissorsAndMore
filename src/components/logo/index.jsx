import React from 'react'
import logoSVG from '../../assets/images/logo-bonus.svg'

import { LogoWrapper } from './styles'

export const Logo = () => (
  <LogoWrapper aria-label="Logo Rock, Paper, Scissors And More">
    <img src={logoSVG} alt="Logo Rock, Paper, Scissors And More"/>
  </LogoWrapper>
)