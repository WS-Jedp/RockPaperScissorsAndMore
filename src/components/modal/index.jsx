import React from 'react'
import ReactDOM from 'react-dom'
import { ModalWrapper } from './styles'

export const Modal = ({children}) => ReactDOM.createPortal((
  <ModalWrapper>
    { children || 'Hello World' }
  </ModalWrapper>), 
  window.document.getElementById('modal-portal'))
