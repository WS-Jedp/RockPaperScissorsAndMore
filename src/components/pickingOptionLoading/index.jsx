import React from 'react'
import { Option } from '../option'

import { PickedOptionWrapper, LoadingWrapper } from './styles'

export const PickingOptionLoading = ({title = 'The House Is Picking...'}) => (
  <PickedOptionWrapper>
    <LoadingWrapper>
      <div></div>
    </LoadingWrapper>
    <strong>
      {title}
    </strong>
  </PickedOptionWrapper>
)