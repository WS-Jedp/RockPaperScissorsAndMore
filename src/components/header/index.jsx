import React from 'react'
import { WrapperHeader } from './styles'

import { Score } from '../score/index'
import { Logo } from '../logo'

export const Header = () => (
  <WrapperHeader>
    <Logo />
    <Score />
  </WrapperHeader>
) 