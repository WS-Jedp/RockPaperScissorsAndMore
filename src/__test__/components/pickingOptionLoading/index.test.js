import React from 'react'
import { mount } from 'enzyme'
import { create } from 'react-test-renderer'

import { PickingOptionLoading } from '../../../components/pickingOptionLoading'

describe('Tests for the "pickingOptionLoading" component', () => {

  describe('Visual tests for the component', () => {
    const PickingOptionLoadingComponent = create(<PickingOptionLoading />) 
    test('should the component be the same', () => {
      expect(PickingOptionLoadingComponent.toJSON()).toMatchSnapshot()
    })
  })

  describe('Functional tests for the component', () => {

    const PickingOptionLoadingMount = mount(<PickingOptionLoading />)

    test('should the compnent return 1 of length', () => {
      expect(PickingOptionLoadingMount.length).toBe(1)
    })

    test('should the title be a string', () => {
      expect(typeof PickingOptionLoadingMount.find('strong').text()).toBe('string')
    })
    
    test('should the title be "the house is picking"', () => {
      expect(PickingOptionLoadingMount.find('strong').text()).toMatch('The House Is Picking...')
    })
  })

})
