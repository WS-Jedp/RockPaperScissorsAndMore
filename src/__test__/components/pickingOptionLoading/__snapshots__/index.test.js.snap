// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for the "pickingOptionLoading" component Visual tests for the component should the component be the same 1`] = `
<article
  className="sc-AxirZ ceaEUY"
>
  <div
    className="sc-AxiKw hBByRz"
  >
    <div />
  </div>
  <strong>
    The House Is Picking...
  </strong>
</article>
`;
